astropy/modeling/functional_models.py

```diff
@@ -400,13 +400,15 @@ def evaluate(x, factor):
         return factor * x
 
 
-class Redshift(Fittable1DModel):
+class RedshiftWavelength(Fittable1DModel):
     """
     One dimensional redshift model.
 
+    Redshifts a wavelength array from rest (z=0) to the input redshift.
+
     Parameters
     ----------
-    z : float or a list of floats
+    z : int or float
```

Comment:
@embray OK, I will remove `int` (I generally like to be explicit).

Perhaps I'm wrong, but I don't think `array-like` `z` is useful here.  For 1D arrays, that means the `x` has to a scalar or match the length of the `z` array (which would then be applied element-wise to `x`).  Is that really useful, e.g. `Redshift(z=[1, 2])(5)` (applying both `z`'s to 5) or `Redshift(z=[1, 2])([5, 10])` (applying z=1 to 5 and z=2 to 10)?